src/libmeasurement_kit/neubot/dash_impl.hpp

```diff
@@ -0,0 +1,345 @@
+            {"Authorization", auth_token},
+        },
+        "", [=](Error error, Var<http::Request> req) {
+            if (error) {
+                logger->warn("dash: request failed: %s",
+                             error.explain().c_str());
+                cb(error);
+                return;
+            }
+            assert(!!req);
+            http_request_recv_response(
+                txp,
+                [=](Error error, Var<http::Response> res) {
+                    if (error) {
+                        logger->warn("dash: cannot receive response: %s",
+                                     error.explain().c_str());
+                        cb(error);
+                        return;
+                    }
+                    assert(!!res);
+                    if (res->status_code != 200) {
+                        logger->warn("dash: invalid response code: %s",
+                                     error.explain().c_str());
+                        cb(http::HttpRequestFailedError());
+                        return;
+                    }
+                    res->request = req;
+                    double length = res->body.length();
+                    double time_elapsed = mk::time_now() - saved_time;
+                    if (time_elapsed <= 0) { // For robustness
+                        logger->warn("dash: invalid time error");
+                        cb(InvalidTimeError());
+                        return;
+                    }
+                    // TODO: we should fill all the required fields
+                    (*entry)["receiver_data"].push_back(report::Entry{
+                        //{"connect_time", self.rtts[0]}
+                        //{"delta_user_time", delta_user_time}
+                        //{"delta_sys_time", delta_sys_time}
+                        {"elapsed", time_elapsed},
+                        {"elapsed_target", DASH_SECONDS},
+                        //{"internal_address", stream.myname[0]}
+                        {"iteration", iteration},
+                        //{"platform", sys.platform}
+                        {"rate", rate_kbit},
+                        //{"real_address", self.parent.real_address}
+                        //{"received", received}
+                        //{"remote_address", stream.peername[0]}
+                        //{"request_ticks", self.saved_ticks}
+                        //{"timestamp", mk::time_now}
+                        //{"uuid", self.conf.get("uuid")}
+                        {"version", MEASUREMENT_KIT_VERSION}});
+                    double speed = length / time_elapsed;
+                    double s_k = (speed * 8) / 1000;
+                    logger->info("[%d/%d] rate: %d kbit/s, speed: %.2f "
+                                 "kbit/s, elapsed: %.2f s",
+                                 iteration, DASH_MAX_ITERATIONS, rate_kbit, s_k,
+                                 time_elapsed);
+                    if (time_elapsed > DASH_SECONDS) {
```

Comment:
This part where we scale down the speed significantly when we think that there was congestion had been criticised during the review and was introduced under the assumption that Neubot was a background tool, but there are also use cases where we run in the foreground. I think it should be on by default to be compatible with Neubot but, moving forward, it should probably be off by default and we should have a bit into the results that indicates whether this was ON or OFF.